src/pyfaf/repos/dnf.py

```diff
@@ -87,34 +94,38 @@ def list_packages(self, architectures):
         release, arch, srpm_name, type, filename, url items.
         """
 
-        self.yum_base.arch.archlist = architectures
-
         result = []
         try:
-            packagelist = self.yum_base.doPackageLists('all', showdups=True)
-        except yum.Errors.RepoError as err:
+            packagelist = self.dnf_base.sack.query().filterm(arch=architectures)
+        except dnf.exceptions.RepoError as err:
             self.log_error("Repository listing failed: '{0}'".format(err))
             return result
 
-        pkgs = (packagelist.available +
-                packagelist.old_available +
-                packagelist.reinstall_available)
+# Get available packages
+###############################################################################
+# TODO: yum added also packagelist.old_available and
+# packagelist.reinstall_available which is missing in dnf
+###############################################################################
+        pkgs = packagelist.available()
 
         for package in pkgs:
             pkg = dict(name=package.name,
-                       base_package_name=package.base_package_name,
                        epoch=package.epoch,
                        version=package.version,
                        release=package.release,
                        arch=package.arch,
-                       url=package.remote_url,
-                       filename=os.path.basename(package.remote_url))
+                       url=package.url,
+                       filename=os.path.basename(package.location))
+###############################################################################
+# The HACK to ensure that pkg["url"] contains an url for the local package,
+# that is missing in dnf
+            if pkg["url"] is None:
+                pkg["url"] = "/".join([str(package.repo.baseurl[0]), os.path.basename(package.location)])
+# End of the hack
+###############################################################################
 
             pkg["type"] = "rpm"
 
             result.append(pkg)
 
-        self.yum_base.closeRpmDB()
-        self.yum_base.close()
```

Comment:
Shouldn't we clean after ourselves? :) 
from documentation to dnf.Base :`To properly finalize and close off any handles the object may hold, client code should either call Base.close() when it has finished operations with the instance, or use the instance as a context manager`